### PR TITLE
[MeshModels] Add Core Application component in MeshModel

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
**Notes for Reviewers**

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Abhishek-kumar09** [comment]
Some meshery-extensions tests were failing in absence of this component, and this regression was quickly picked.

File: server/meshmodel/components/core/v1.0.0/Application.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+	"kind": "Application",
+	"apiVersion": "core.oam.dev/v1alpha1",
```
Comment (leecalcote) [inline server/meshmodel/components/core/v1.0.0/Application.json:3]: @Revolyssup oam?

**Review by leecalcote — CHANGES_REQUESTED** [review]
Needs to be of type MeshModel

File: server/meshmodel/components/core/v1.0.0/Application.json (hunk 1 above)
Comment (Abhishek-kumar09) [reply server/meshmodel/components/core/v1.0.0/Application.json:3]:
```suggestion
	"apiVersion": "core.meshmodel.dev/v1alpha1",
```

**Review by Revolyssup — APPROVED** [review]

**Pull request merged**